src/Builder/QueryType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder;
+
+use MongoDB\Builder\Query\TypeOperator;
+use MongoDB\Builder\Type\Encode;
+use MongoDB\Builder\Type\FieldQueryInterface;
+use MongoDB\Builder\Type\OperatorInterface;
+
+/**
+ * Shortcut for $type field query operator
+ *
+ * @see https://www.mongodb.com/docs/manual/reference/operator/query/type/#available-types
+ * @see TypeOperator
+ */
+enum QueryType: string implements OperatorInterface, FieldQueryInterface
+{
+    public const ENCODE = Encode::Array;
+
+    case Double = 'double';
+    case String = 'string';
+    case Object = 'object';
+    case Array = 'array';
+    case BinaryData = 'binData';
+    case ObjectId = 'objectId';
+    case Boolean = 'bool';
+    case Date = 'date';
+    case Null = 'null';
+    case Regex = 'regex';
+    case JavaScript = 'javascript';
+    case Int = 'int';
+    case Timestamp = 'timestamp';
+    case Long = 'long';
```

Comment:
```suggestion
    case Int64 = 'long';
```

This also leaves nothing to the imagination.

Reply:
It just occurred to me that you might have chosen these names to be consistent with the ExpressionFactoryTrait field path methods. Unless it was just the first column from the table in  [$type: Available Types](https://www.mongodb.com/docs/manual/reference/operator/query/type/#available-types).

In any event, you may want to keep the factory methods consistent with the enums here. And if factory methods will be distinguishing 32-bit and 64-bit integers, I do think explicit names with the bit size are more clear to users than "int" and "long".

Reply:
I updated all the type name to take more descriptive ones, from the BSON classes when they exist.